### PR TITLE
TSL: Add function for BPCEM.

### PR DESCRIPTION
**Pull request by Mugen87**
Related issue: #15897

**Description**

This PR introduces a helper function for using  box projected cube environment mapping (BPCEM) for cube textures and PMREMs. This is a nice feature when using environment maps in indoor scenes.

See https://devlog-martinsh.blogspot.com/2011/09/box-projected-cube-environment-mapping.html.

We had this feature in the repository in earlier days as an example but eventually removed it since the shader patching approach was error-prone and broke the demo more than once. Now with TSL we can implement this feature in a much more elegant fashion. There is now the TSL function`getParallaxCorrectNormal()` which can be used to compute custom uvs when using `cubeTexture()` or `pmremTexture()`.

### CODE REVIEW

**Comment by sunag** [comment]
Wow! It was very easy to use now.

File: examples/files.json [hunk 1]
```diff
@@ -354,6 +354,7 @@
 		"webgpu_materials_arrays",
 		"webgpu_materials_basic",
 		"webgpu_materials_displacementmap",
+		"webgpu_materials_envmaps_bpcem",
```
Comment (WestLangley) [inline examples/files.json:357]: How about a different name, for clarity?

```js
webgpu_materials_envmaps_box_projected
```
Similar to
```js
webgl_materials_envmaps_ground_projected
```
(The latter is currently missing an underscore.)

Comment (Mugen87) [reply examples/files.json:357]: > webgl_materials_envmaps_ground_projected

I think this should be `webgl_skybox_ground_projected`. It's not really about environment mapping...

Comment (WestLangley) [reply examples/files.json:357]: Whatever you prefer is OK with me... I thought the acronym may be unfamiliar.

**Pull request merged**